SDK/MobileCenterPush/Microsoft.Azure.Mobile.Push.UWP/Push.cs

```diff
@@ -1,67 +1,174 @@
+                        e.Cancel = true;
+                        PlatformPushNotificationReceived?.Invoke(sender, pushNotification);
+                        MobileCenterLog.Debug(LogTag, "Application in foreground. Intercept push notification and invoke push callback.");
+                    }
+                    else
+                    {
+                        MobileCenterLog.Debug(LogTag, $"Push ignored. It was not sent through Mobile Center.");
+                    }
+                }
             }
             else
             {
-                MobileCenterLog.Error(LogTag, "Push service registering with Mobile Center backend has failed.");
+                MobileCenterLog.Debug(LogTag, $"Push ignored. We only handle Toast notifications but PushNotificationType is '{e.NotificationType}'.");
             }
-
-            _mutex.Unlock();
         }
 
-        private async Task<PushNotificationChannel> CreatePushNotificationChannel()
+        private PushNotificationReceivedEventArgs ParseMobileCenterPush(XmlDocument content)
         {
-            PushNotificationChannel channel = await new WindowsPushNotificationChannelManager().CreatePushNotificationChannelForApplicationAsync();
+            // Check if mobile center push (it always has launch attribute with JSON object having mobile_center key
+            var launch = content.SelectSingleNode("/toast/@launch")?.NodeValue.ToString();
+            var customData = ParseLaunchString(launch);
+            if (customData == null)
+            {
+                return null;
+            }
 
-            return channel;
+            // Parse title and message using identifiers
+            return new PushNotificationReceivedEventArgs()
+            {
+                Title = content.SelectSingleNode("/toast/visual/binding/text[@id='1']")?.InnerText,
+                Message = content.SelectSingleNode("/toast/visual/binding/text[@id='2']")?.InnerText,
+                CustomData = customData
+            };
         }
 
-        private static event EventHandler<PushNotificationReceivedEventArgs> PlatformPushNotificationReceived;
+        private static Dictionary<string, string> ParseLaunchString(string launchString)
+        {
+            try
+            {
+                if (launchString != null)
+                {
+                    var launchJObject = JObject.Parse(launchString);
+                    if (launchJObject?["mobile_center"] is JObject mobileCenterData)
+                    {
+                        var customData = new Dictionary<string, string>();
+                        foreach (var pair in mobileCenterData)
+                        {
+                            customData.Add(pair.Key, pair.Value.ToString());
+                        }
+                        return customData;
+                    }
+                }
+                return null;
+            }
+            catch (JsonReaderException)
+            {
+                return null;
```

Comment:
In general, would it not be better to ignore these low level exceptions and have them caught at the highest level possible so that we can eliminate many of the null checks and delegate the error control flow logic to the built in exception mechanisms? Not sure what the right answer is